account_export_csv/wizard/account_export_csv.py

```diff
@@ -0,0 +1,367 @@
+            _("DEBIT"),
+            _("CREDIT"),
+            _("BALANCE"),
+        ]
+
+    def _get_rows_analytic(self, journal_ids):
+        """
+        Return list to generate rows of the CSV file
+        """
+        self.ensure_one()
+        self.env.cr.execute(
+            """  select aac.code as analytic_code,
+                        aac.name as analytic_name,
+                        ac.code,ac.name,
+                        sum(debit) as sum_debit,
+                        sum(credit) as sum_credit,
+                        sum(debit) - sum(credit) as balance
+                        from account_move_line
+                        left outer join account_analytic_account as aac
+                        on (account_move_line.analytic_account_id = aac.id)
+                        inner join account_account as ac
+                        on account_move_line.account_id = ac.id
+                        AND account_move_line.date >= %(date_start)s
+                        AND account_move_line.date <= %(date_end)s
+                        group by aac.id,aac.code,aac.name,ac.id,ac.code,ac.name
+                        order by aac.code
+                   """,
+            {"date_start": self.date_start, "date_end": self.date_end},
+        )
+        res = self.env.cr.fetchall()
+
+        rows = []
+        for line in res:
+            rows.append(list(line))
+        return rows
+
+    def action_manual_export_journal_entries(self):
+        """
+        Here we use TemporaryFile to avoid full filling the Odoo worker
+        Memory
+        We also write the data to the wizard with SQL query as write seems
+        to use too much memory as well.
+        Those improvements permitted to improve the export from a 100k line to
+        200k lines
+        with default `limit_memory_hard = 805306368` (768MB) with more lines,
+        you might encounter a MemoryError when trying to download the file even
+        if it has been generated.
+        To be able to export bigger volume of data, it is advised to set
+        limit_memory_hard to 2097152000 (2 GB) to generate the file and let
+        Odoo load it in the wizard when trying to download it.
+        Tested with up to a generation of 700k entry lines
+        """
+        self.ensure_one()
+        rows = self._get_data("journal_entries")
+        with tempfile.TemporaryFile() as file_data:
+            writer = AccountingWriter(file_data)
+            writer.writerows(rows)
+            with tempfile.TemporaryFile() as base64_data:
+                file_data.seek(0)
+                base64.encode(file_data, base64_data)
+                base64_data.seek(0)
+                self.write({"data": base64_data.read()})
+                # self.env.cr.execute(
+                #     """
+                # UPDATE account_csv_export
+                # SET data = %s
+                # WHERE id = %s""",
+                #     (base64_data.read(), self.id),
+                # )
```

Comment:
Yes @Kev-Roche could you reset the older commented code, as suggested by @StefanRijnhart. Thanks

Reply:
@bealdav @StefanRijnhart I reset the older code, but as you can see, it brings an error, I don't understand why.